### PR TITLE
Added Go 1.14.6 and 1.13.14

### PR DESCRIPTION
**Pull request by paulbry**
Added another Go release (1.14.6 & 1.13.14) - https://golang.org/doc/devel/release.html#go1.14.minor

Also contains the support for `spack external find go`

### CODE REVIEW

**Comment by adamjstewart** [comment]
New version looks good. #17804 will greatly simplify the external package detection syntax. Would you like to wait until it is merged and update this, or merge and update later?

**Comment by paulbry** [comment]
@adamjstewart sure thing, I've gone ahead and removed it. Looking forward to that PR being merge.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**
